packages/dashboards/tests/acceptance/dashboards-test.js

```diff
@@ -433,4 +450,52 @@ module('Acceptance | Dashboards', function(hooks) {
         'Unauthorized widget loaded unauthorized component'
       );
   });
+
+  test('dashboard save/revert', async function(assert) {
+    assert.expect(10);
+
+    server.patch('/dashboards/1', () => {
+      assert.notok(true, 'Dashboard should not be patched before save.');
+      return new Response(500);
+    });
+
+    await visit('/dashboards/1');
+
+    assert
+      .dom('.navi-dashboard__revert-button')
+      .isNotVisible('Revert button is not visible before making a change to the dashboard');
+
+    await click('.editable-label__icon');
+    await fillIn('.editable-label__input', 'ABCD');
+    await blur('.editable-label__input');
+
+    assert.dom('.navi-dashboard__edit-title').hasText('ABCD', 'Has updated text');
+
+    assert.dom('.navi-dashboard__revert-button').isVisible('Revert button appears after dashboard change');
+
+    await click('.navi-dashboard__revert-button');
+
+    assert.dom('.navi-dashboard__revert-button').isNotVisible('Revert button is not visible after reverting');
+
+    assert.dom('.navi-dashboard__edit-title').hasText('Tumblr Goals Dashboard', 'Has original text');
+
+    assert.dom('.navi-dashboard__save-button').isNotVisible('Save should not be visible after revert');
+
+    await click('.editable-label__icon');
+    await fillIn('.editable-label__input', 'EFGH');
+    await blur('.editable-label__input');
+
+    assert.dom('.navi-dashboard__save-button').isVisible('Save appears after change');
+
+    server.patch('/dashboards/1', () => {
+      assert.ok(true, 'Dashboard should be patched after save.');
+      return new Response(500);
```

Comment:
Do we want to return a 500 here?

Reply:
The app freaks out if a 200 returns with no body.